app/src/main/java/com/example/atheneum/activities/ScanBarcodeActivity.java

```diff
@@ -0,0 +1,162 @@
+
+                        Log.i(TAG, "I have permission");
+
+                        // Permission has already been granted
+                        try {
+                            cameraSource.start(cameraPreview.getHolder());
+                        } catch (IOException e) {
+                            e.printStackTrace();
+                        }
+                        catch(SecurityException e){
+                            e.printStackTrace();
+                        }
+
+                    Log.i(TAG, "surface created");
+                }
+
+                @Override
+                public void surfaceChanged(SurfaceHolder holder, int format, int width, int height) {
+
+                }
+
+                @Override
+                public void surfaceDestroyed(SurfaceHolder holder) {
+                    cameraSource.stop();
+                }
+            });
+        }
+        barcodeDetector.setProcessor(new Detector.Processor<Barcode>() {
+            @Override
+            public void release() {
+
+            }
+
+            @Override
+            public void receiveDetections(Detector.Detections<Barcode> detections) {
+                Log.i(TAG, "in receivedDetections");
+                final SparseArray<Barcode> barcode = detections.getDetectedItems();
+                if(barcode.size() != 0){
+                    Intent intent = new Intent();
+                    intent.putExtra("Barcode", barcode.valueAt(0));
+                    setResult(CommonStatusCodes.SUCCESS, intent);
+                    finish();
+                }
+            }
+        });
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
+        switch (requestCode) {
+            case MY_PERMISSIONS_REQUEST_CAMERA: {
+                Log.i(TAG, "in onRequestPermissionResult");
+                // If request is cancelled, the result arrays are empty.
+                if (grantResults.length > 0
+                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    // permission was granted, yay! Do the
+                    // contacts-related task you need to do.
+                    if (ActivityCompat.checkSelfPermission(this, Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED)
+                        Log.i(TAG, "in onRequestPermissionResult, has permission");
+                        try {
+                            Log.i(TAG, "in onRequestPermissionResult, start camera");
+                            cameraSource.start(cameraPreview.getHolder());
+                        } catch (IOException e) {
+                            e.printStackTrace();
+                        }
+                        catch(SecurityException e){
+                            e.printStackTrace();
+                        }
+                } else {
+                    // permission denied, boo! Disable the
```

Comment:
Are you sure it's okay to leave this blank? Maybe add a log here.

Reply:
Ok, I've added a log statement